java/arcs/core/util/testutil/LogRule.kt

```diff
@@ -27,7 +28,7 @@ class LogRule : TestRule {
 
     override fun apply(base: Statement, desc: Description): Statement = object : Statement() {
         override fun evaluate() {
-            val messages = mutableListOf<String>()
+            val messages = CopyOnWriteArrayList<String>()
```

Comment:
Just out of curiosity: why changed to CopyOnWriteArrayList? For multi-threading?

Reply:
Yeah, I was noticing a bizarre race condition in test cases where things would crash  due to edits being made concurrently to this list - and the expected data from the list not being available.